File contents

BobGu opened the pull request:
This feature allows a user to click on a file name, if it is a text file then they are allowed to see those contents.  These files are read only, so a method not allowed is returned for request that are non read(post, put, delete etc..)

Commit:
File one controller allows resource to be read and returns a string

Commit:
Renamed File one controller to file controller, which handles file1 file 2 and text-file paths.

Changed parser reggae to include dashes and periods.

Added new routes

jayden commented on src/main/java/Routes/Router.java:
```diff
@@ -55,6 +55,9 @@ private void createRoutes() {
         routes.add(new Route("/redirect", new RedirectController()));
+        routes.add(new Route("/file1", new FileController(new FileReader())));
+        routes.add(new Route("/file2", new FileController(new FileReader())));
+        routes.add(new Route("/text-file.txt", new FileController(new FileReader())));
```
Be careful not to tie your implementation to the CobSpec tests. The server should be able to serve requests for any file and any directory.

Commit:
Request has a field title isFile

Parser parses the path, then checks directory name, and if file exists
using directory name and path then boolean is assigned to the request
object

Commit:
Request has isImage field

Parser contains image, sees if it is a file, if it is a file then it
checks to see if it has any image extensions and returns true or false.

Commit:
Image files correctly get encoded and dispalyed as data url links

Then they are formatted into html and sent as a string back, correctly
displays images on the browser.

Commit:
WIP need make a response object that has body and headers, that way I can return byte array for the image and not a string

Commit:
WIP changing controller handle return from string to byte array

Commit:
Fixed file controller and form controller to return correct byte array as response

Commit:
Fixed failing tests
